roles/tas_single_node/tasks/podman.yml

```diff
@@ -12,10 +12,17 @@
     - "{{ tas_single_node_kube_manifest_dir }}"
     - "{{ tas_single_node_kube_configmap_dir }}"
 
+- name: Podman check login
+  ansible.builtin.command: podman login --get-login registry.redhat.io
+  register: logincheck
+  failed_when: logincheck.stdout == "Nil val"
+  changed_when: logincheck.stdout | string != tas_single_node_registry_username | string
+
 - name: Podman login to registry.redhat.io
   ansible.builtin.command: podman login registry.redhat.io -u '{{ tas_single_node_registry_username }}' --password {{ tas_single_node_registry_password }}
   register: podman_login_result
-  changed_when: '"Already logged in" not in podman_login_result'
+  when: logincheck.stdout | string != tas_single_node_registry_username | string
+  changed_when: '"Login Succeeded!" in podman_login_result.stdout'
```

Comment:
When we run this task, we would always expect it to be changed (because it changes the state of the machine from not being logged in to being logged in), so the `changed_when` can be removed altogether I think.

Reply:
This task is always expected to be changed?
If so i can get rid of the check before hand in that case no?

Reply:
@tommyd450 this task is always expected to be changed *when it is run*. We just don't want to run it when we're already logged in. Does that make sense?

Reply:
Right gotcha,